### PR TITLE
feat: add out of budget to proposal status

### PR DESCRIPTION
**Pull request by 1emu**
Not sure if we should be able to link grant proposals with OOB status, but if the idea is to be able to resubmit and link to them, I think we should.

Moved frontend behavior depending on `ProposalStatus` to tested functions in `utils.ts`, so it's easier to refactor them later.

### CODE REVIEW

File: src/entities/Proposal/utils.ts [hunk 1]
```diff
@@ -163,3 +149,39 @@ export function isProposalInCliffPeriod(grant: TransparencyGrant) {
...
+    default:
+      return false
+  }
+}
+
+export function toProposalStatus(value: string | null | undefined, orElse: () => any): ProposalStatus | any {
+  return isProposalStatus(value) ? (value as ProposalStatus) : orElse()
+}
+
+export function proposalCanBeDeleted(proposalStatus?: ProposalStatus) {
```
Comment (andyesp) [inline src/entities/Proposal/utils.ts:173]: `isProposalDeletable`?

File: src/entities/Proposal/utils.ts [hunk 2]
```diff
@@ -163,3 +149,39 @@ export function isProposalInCliffPeriod(grant: TransparencyGrant) {
...
+  return proposalStatus === ProposalStatus.Pending || proposalStatus === ProposalStatus.Active
+}
+
+export function proposalCanBeEnacted(proposalStatus?: ProposalStatus) {
```
Comment (andyesp) [inline src/entities/Proposal/utils.ts:177]: `isProposalEnactable`?

File: src/entities/Proposal/templates/messages.ts [hunk 3]
```diff
@@ -2,10 +2,15 @@ import { Vote } from '../../Votes/types'
 import { calculateResult } from '../../Votes/utils'
 import { ProposalAttributes, ProposalStatus } from '../types'
 
+function getProposalStatusDisplayName(proposalStatus: ProposalStatus) {
+  if (proposalStatus === ProposalStatus.OutOfBudget) return 'OUT OF BUDGET'
```
Comment (andyesp) [inline src/entities/Proposal/templates/messages.ts:6]: You can do something like `ProposalStatus.OutOfBudget.split('_').join(' ').toUpperCase()`

File: src/intl/en.json [hunk 4]
```diff
@@ -286,7 +286,8 @@
     "active": "Active outcomes",
     "rejected": "Rejected outcomes",
     "passed": "Passed outcomes",
-    "enacted": "Enacted outcomes"
+    "enacted": "Enacted outcomes",
+    "out_of_budget": "OOB outcomes"
```
Comment (andyesp) [inline src/intl/en.json:290]: Is this ok with @ginocingolani? Not sure about the acronym here.

Comment (1emu) [reply src/intl/en.json:290]: I believe Guane is going to use the acronym in the pills. But we can change it later :)

**Pull request merged**